.github/workflows/_test_matrix.yml

```diff
@@ -0,0 +1,32 @@
+on:
+  workflow_call:
+    outputs:
+      matrix: 
+          value: ${{ jobs.test_matrix.outputs.matrix }}
+
+env:
+  WORK_DIR: /__w/PandABlocks-FPGA/PandABlocks-FPGA
+
+jobs:
+    test_matrix:
+      runs-on:
+        group: iris_runners
+      container:
+          image: docker.io/shihabdls/pandablocks-container-extension:v2.5
+          options: --privileged
+      outputs:
+          matrix: ${{steps.make_matrix.outputs.matrix}} 
+      steps:
+        - name: Checkout Source
+          uses: actions/checkout@v4
+          with:
+            path: repos/PandABlocks-fpga
+            fetch-depth: 0
+  
+        - name: Evaluate number of modules/tests & generate job matrix
+          id: make_matrix
+          run: |
+               bash $WORK_DIR/repos/PandABlocks-fpga/.github/scripts/Find_Tests.sh "$WORK_DIR/repos/PandABlocks-fpga/modules" "$WORK_DIR/repos/PandABlocks-fpga/.github/scripts"
+               json_content=$(cat github_tests.json)
+               echo "::set-output name=matrix::$json_content"
+               echo "$json_content"
```

Comment:
Is this finding of tests and separating into different runners via the matrix needed with the cocotb tests?

Reply:
No, this was done to reduce the time taken for the testbenches. It was taking ~4 hours for all of the tests to complete as Vivado's simulation was single threaded, so @shihab-dls made a script to separate them into 4 roughly equivalent sized batches of tests so they can be run in parallel on different runners. However, the whole cocotb test only takes a few minutes.

Reply:
Ok, happy to merge this as is then simplify later with the cocotb changes